### PR TITLE
Preparation of the Wx attachment implementation

### PR DESCRIPTION
**Pull request by rafaelx**

### CODE REVIEW

File: src/ui/wxWidgets/MenuEditHandlers.cpp [hunk 1]
```diff
@@ -53,28 +53,37 @@
...
+      this, m_core,
+      isItemReadOnly ? AddEditPropSheetDlg::SheetType::VIEW : AddEditPropSheetDlg::SheetType::EDIT,
+      &item
+    );
+
+    returnCode = dialog.ShowModal();
   }
-  if (rc == wxID_OK) {
+  else {
+    EditShortcutDlg dialog(this, m_core, &item);
```
Comment (ronys) [inline src/ui/wxWidgets/MenuEditHandlers.cpp:76]: EditShortcutDlg should also have r/w and r-o variants.

File: src/core/ItemAtt.h [hunk 2]
```diff
@@ -38,6 +38,10 @@ class PWSfile;
 class CItemAtt : public CItem
 {
 public:
+  // The maximum supported attachment size
+  // according to section 3.4 of formatV4 specification.
+  const static size_t MAX_SIZE = 4294967296; // 2^32
```
Comment (ronys) [inline src/core/ItemAtt.h:43]: Not sure how this would work on a 32bit compilation

File: src/ui/wxWidgets/wxUtilities.cpp [hunk 3]
```diff
@@ -215,3 +215,213 @@ bool IsTaskBarIconAvailable()
...
+  Unbind(wxEVT_PAINT, &ImagePanel::OnPaint, this);
+  Unbind(wxEVT_SIZE, &ImagePanel::OnSize, this);
+}
+
+bool ImagePanel::LoadFromFile(const wxString &file, wxBitmapType format)
+{
+  if (m_Image.LoadFile(file, format)) {
+
+    DetermineImageProperties(m_Image);
+    Refresh(); // Triggers OnPaint to display the image
```
Comment (ronys) [inline src/ui/wxWidgets/wxUtilities.cpp:240]: I understand that this is where the size problem occurs? If so, did you try refreshing the parent window?

File: src/ui/wxWidgets/wxUtilities.cpp [hunk 4]
```diff
@@ -215,3 +215,213 @@ bool IsTaskBarIconAvailable()
...
+    if (newHeight < 1) {
+      newHeight = 1;
+    }
+
+    // Scale the image and show it with its new dimensions
+    m_Bitmap = wxBitmap(m_Image.Scale(newWidth, newHeight));
+    DrawBitmapCentered(dc, wxSize(areaWidth, areaHeight), wxSize(newWidth, newHeight));
+  }
+  else {
+    // No scaling needed, so we use the image with its original dimensions
```
Comment (ronys) [inline src/ui/wxWidgets/wxUtilities.cpp:415]: Given the return statement in line 398, will this 'else' clause ever be called?

File: src/ui/wxWidgets/MenuEditHandlers.cpp (hunk 1 above)
Comment (rafaelx) [reply src/ui/wxWidgets/MenuEditHandlers.cpp:76]: I did only some code formatting and haven't had a look at the functionality of the shortcut editing dialog. I will do this later on a separate branch. Currently, in R-O mode the dialog doesn't allow modifications to a shortcut.

File: src/ui/wxWidgets/wxUtilities.cpp (hunk 3 above)
Comment (rafaelx) [reply src/ui/wxWidgets/wxUtilities.cpp:240]: I have not yet found a suitable solution.

File: src/core/ItemAtt.h [hunk 5]
```diff
@@ -40,7 +40,7 @@ class CItemAtt : public CItem
 public:
   // The maximum supported attachment size
   // according to section 3.4 of formatV4 specification.
-  const static size_t MAX_SIZE = 4294967296; // 2^32
+  const static size_t MAX_SIZE = 4294967295U; // 2^32
```
Comment (rafaelx) [inline src/core/ItemAtt.h:43]: See https://en.cppreference.com/w/cpp/language/types and https://rules.sonarsource.com/c/RSPEC-854

**Pull request merged**